### PR TITLE
feat: fetch announcements for user

### PR DESCRIPTION
**Pull request by marcomariscal**
when parsing/processing events: need to figure out how to make doing many computations/making many requests to find the "from" value of the tx more performant

### CODE REVIEW

File: examples/getAnnouncementsForUser/.env.example [hunk 1]
```diff
@@ -0,0 +1 @@
+RPC_URL='Your rpc url'
```
Comment (garyghayrat) [inline examples/getAnnouncementsForUser/.env.example:1]: Is this right?
```suggestion
RPC_URL='Your rpc url'
TEST_STEALTH_META_ADDRESS_URI=
TEST_SPENDING_PUBLIC_KEY=
TEST_SPENDING_PRIVATE_KEY=
TEST_VIEWING_PUBLIC_KEY=
TEST_VIEWING_PRIVATE_KEY=
```

Comment (garyghayrat) [reply examples/getAnnouncementsForUser/.env.example:1]: If it is, also need to update the main `.env.example`

File: examples/getAnnouncementsForUser/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    fromBlock, // Optional fromBlock parameter (defaults to 0, which can be slow for many blocks)
+    toBlock: 'latest', // Optional toBlock parameter (defaults to latest)
+  });
+
+  // Example call to getAnnouncementsForUser action on the stealth client
+  // Adjust parameters according to your requirements
+  const userAnnouncements = await stealthClient.getAnnouncementsForUser({
+    announcements,
+    spendingPublicKey: '0xYourSpendingPublicKey',
+    viewingPrivateKey: '0xYourViewingPrivateKey',
```
Comment (garyghayrat) [inline examples/getAnnouncementsForUser/index.ts:45]: Replace w/ `process.env.`

Comment (marcomariscal) [reply examples/getAnnouncementsForUser/index.ts]: https://github.com/ScopeLift/stealth-address-sdk/pull/22/commits/ea953ca5de072f1e6ac640fcefc1b333ed88cef9

**Review by garyghayrat — APPROVED** [review]
Still need to update the root `.env.example` mentioned here https://github.com/ScopeLift/stealth-address-sdk/pull/22#discussion_r1505008211, otherwise looks good to me.

File: examples/getAnnouncementsForUser/.env.example (hunk 1 above)
Comment (marcomariscal) [reply examples/getAnnouncementsForUser/.env.example:3]: removed the irrelevant test env vars

**Comment by marcomariscal** [comment]
> Still need to update the root `.env.example` mentioned here [#22 (comment)](https://github.com/ScopeLift/stealth-address-sdk/pull/22#discussion_r1505008211), otherwise looks good to me.

src and the main directory don't use `RPC_URL`, so will keep as is for now.

**Pull request merged**